tests/integration/clientv3/metrics_test.go

```diff
@@ -168,11 +170,186 @@ func getHTTPBodyAsLines(t *testing.T, url string) []string {
 		if err != nil {
 			if errors.Is(err, io.EOF) {
 				break
+			} else {
+				t.Fatalf("error reading: %v", err)
 			}
-			t.Fatalf("error reading: %v", err)
 		}
 		lines = append(lines, line)
 	}
 	resp.Body.Close()
 	return lines
 }
+
+func TestAllMetricsGenerated(t *testing.T) {
+	integration2.BeforeTest(t)
+
+	var (
+		addr = "localhost:27989"
+		ln   net.Listener
+	)
+
+	srv := &http.Server{Handler: promhttp.Handler()}
+	srv.SetKeepAlivesEnabled(false)
+
+	ln, err := transport.NewUnixListener(addr)
+	if err != nil {
+		t.Errorf("Error: %v occurred while listening on addr: %v", err, addr)
+	}
+
+	donec := make(chan struct{})
+	defer func() {
+		ln.Close()
+		<-donec
+	}()
+
+	// listen for all Prometheus metrics
+	go func() {
+		defer close(donec)
+
+		serr := srv.Serve(ln)
+		if serr != nil && !transport.IsClosedConnError(serr) {
+			t.Errorf("Err serving http requests: %v", serr)
+		}
+	}()
+
+	url := "unix://" + addr + "/metrics"
```

Comment:
Thinking whether e2e test here would better, more accurate and would not require this custom code. But don't want to block on it.

Reply:
🙏🏾